core/src/main/scala/org/apache/spark/network/netty/NettyBlockTransferService.scala

```diff
@@ -112,8 +117,19 @@ private[spark] class NettyBlockTransferService(
       val blockFetchStarter = new RetryingBlockFetcher.BlockFetchStarter {
         override def createAndStart(blockIds: Array[String], listener: BlockFetchingListener) {
           val client = clientFactory.createClient(host, port)
-          new OneForOneBlockFetcher(client, appId, execId, blockIds, listener,
-            transportConf, tempFileManager).start()
+          try {
+            new OneForOneBlockFetcher(client, appId, execId, blockIds, listener,
+              transportConf, tempFileManager).start()
+          } catch {
+            case e: IOException =>
+              assert(driverEndPointRef != null)
+              if (driverEndPointRef.askSync[Boolean](IsExecutorAlive(execId))) {
```

Comment:
If `askSync` fails, we should throw the original exception.

Reply:
> If `askSync` fails, we should throw the original exception.
Thanks,  I used to think that if askSync fails, it would also throw another IOException, whose stackTrace is similar with origin IOException.
Done.

Reply:
should we remove the `assert(driverEndPointRef != null)`, because the invocation of driverEndPointRef is in `Try`. @cloud-fan